packages/@react-aria/combobox/docs/useComboBox.mdx

```diff
@@ -0,0 +1,1166 @@
+    </div>
+  );
+}
+
+function ListBoxPopup(props) {
+  let {
+    popoverRef,
+    listBoxRef,
+    state,
+    shouldUseVirtualFocus,
+    ...otherProps
+  } = props;
+
+  // Get props for the listbox.
+  // Prevent focus moving to listbox via shouldUseVirtualFocus
+  let {listBoxProps} = useListBox({
+    autoFocus: state.focusStrategy,
+    disallowEmptySelection: true,
+    shouldUseVirtualFocus,
+    ...otherProps
+  }, state, listBoxRef);
+
+  // Handle events that should cause the popup to close,
+  // e.g. blur, clicking outside, or pressing the escape key.
+  let {overlayProps} = useOverlay({
+    onClose: () => state.close(),
+    shouldCloseOnBlur: true,
+    isOpen: state.isOpen,
+    isDismissable: true
+  }, popoverRef);
+
+  // Add hidden <DismissButton> components at the start and end of the list
+  // to allow screen reader users to dismiss the popup easily.
+  return (
+    <div {...overlayProps} ref={popoverRef}>
+      <DismissButton onDismiss={() => state.close()} />
+      <ul
+        {...mergeProps(listBoxProps, otherProps)}
+        ref={listBoxRef}
+        style={{
+          position: 'absolute',
+          width: '100%',
+          margin: '4px 0 0 0',
+          padding: 0,
+          listStyle: 'none',
+          border: '1px solid gray',
+          background: 'lightgray'
+        }}>
+        {[...state.collection].map(item => (
+          <Option
+            shouldUseVirtualFocus
+            key={item.key}
+            item={item}
+            state={state} />
+        ))}
+      </ul>
+      <DismissButton onDismiss={() => state.close()} />
+    </div>
+  )
+}
+
+function Option({item, state, shouldUseVirtualFocus}) {
+  let ref = React.useRef();
+  let isDisabled = state.disabledKeys.has(item.key);
+  let isSelected = state.selectionManager.isSelected(item.key);
+  // Track focus via focusedKey state instead of with focus event listeners
+  // since focus never leaves the text input in a ComboBox
+  let isFocused = state.selectionManager.focusedKey === item.key;
+
+  // Get props for the option element.
+  // Prevent options from receiving browser focus via shouldUseVirtualFocus.
+  let {optionProps} = useOption({
+    key: item.key,
+    isDisabled,
+    isSelected,
+    shouldSelectOnPressUp: true,
+    shouldFocusOnHover: true,
+    shouldUseVirtualFocus
+  }, state, ref);
+
+  let backgroundColor;
+  let color;
+
+  if (isSelected) {
+    backgroundColor = 'blueviolet';
+    color = 'white';
+  } else if (isFocused) {
+    backgroundColor = 'gray';
+    color = 'black';
+  } else if (isDisabled) {
+    backgroundColor = 'transparent';
+    color = 'gray';
+  }
+
+  return (
+    <li
+      {...optionProps}
+      ref={ref}
+      style={{
+        background: backgroundColor,
+        color: color,
+        padding: '2px 5px',
+        outline: 'none',
+        cursor: 'pointer'
+      }}>
+      {item.rendered}
+    </li>
+  );
+}
+
+<ComboBox label="Favorite Animal" defaultSelectedKey="red panda">
```

Comment:
Should this example have an empty dropdown, seen as a line in the image, when nothing matches? 
![image](https://user-images.githubusercontent.com/309540/104532307-c3b06380-55c4-11eb-8dcd-7e3d609d1082.png)

Reply:
Hm, I can't seem to get that to happen. What browser/repro steps?

Reply:
Chrome (that is a version behind) and attaching video. 

https://user-images.githubusercontent.com/309540/104660807-1a2ca900-567c-11eb-808a-9e39debe6afd.mov

Reply:
Ahhh, I see, problem with the controlled story, lemme see if I can change something

Reply:
kk, I pushed up a fix, good catch. I forgot to add items to the field state since that was controlled as well and useComboBoxState doesn't close the menu if the collection is empty and items/open state is controlled, that is up to the user event handlers